plugins/woocommerce-blocks/assets/js/base/components/select/index.tsx

```diff
@@ -22,61 +27,151 @@ type SelectProps = Omit<
+	);
+	const errorIdString = errorId || restOfProps.id || '';
+	const optionsWithEmpty = useMemo< SelectOption[] >( () => {
+		return [ emptyOption ].concat( options );
+	}, [ options, emptyOption ] );
+
+	const { setValidationErrors, clearValidationError } =
+		useDispatch( VALIDATION_STORE_KEY );
+
+	const { error, validationErrorId } = useSelect( ( select ) => {
+		const store = select( VALIDATION_STORE_KEY );
+		return {
+			error: store.getValidationError( errorIdString ),
+			validationErrorId: store.getValidationErrorId( errorIdString ),
+		};
+	} );
+
+	useEffect( () => {
+		if ( ! required || value ) {
+			clearValidationError( errorIdString );
+		} else {
+			setValidationErrors( {
+				[ errorIdString ]: {
+					message: errorMessage,
+					hidden: true,
+				},
+			} );
+		}
+		return () => {
+			clearValidationError( errorIdString );
+		};
+	}, [
+		clearValidationError,
+		value,
+		errorIdString,
+		errorMessage,
+		required,
+		setValidationErrors,
+	] );
+
+	const validationError = useSelect( ( select ) => {
+		const store = select( VALIDATION_STORE_KEY );
+		return (
+			store.getValidationError( errorIdString || '' ) || {
+				hidden: true,
+			}
+		);
+	} );
+
 	const generatedId = useId();
 
 	const inputId =
 		restOfProps.id || `wc-blocks-components-select-${ generatedId }`;
 
 	return (
-		<div className={ `wc-blocks-components-select ${ className || '' }` }>
-			<div className="wc-blocks-components-select__container">
-				<label
-					htmlFor={ inputId }
-					className="wc-blocks-components-select__label"
-				>
-					{ label }
-				</label>
-				<select
-					className="wc-blocks-components-select__select"
-					id={ inputId }
-					size={ size !== undefined ? size : 1 }
-					onChange={ selectOnChange }
-					value={ value }
-					{ ...restOfProps }
-				>
-					{ options.map( ( option ) => (
-						<option
-							key={ option.value }
-							value={ option.value }
-							data-alternate-values={ `[${ option.label }]` }
-							disabled={
-								option.disabled !== undefined
-									? option.disabled
-									: false
-							}
-						>
-							{ option.label }
-						</option>
-					) ) }
-				</select>
-				<Icon
-					className="wc-blocks-components-select__expand"
-					icon={ chevronDown }
-				/>
+		<div
+			className={ clsx( className, {
+				'has-error': ! validationError.hidden,
+			} ) }
+		>
+			<div className="wc-blocks-components-select">
+				<div className="wc-blocks-components-select__container">
+					<label
+						htmlFor={ inputId }
+						className="wc-blocks-components-select__label"
+					>
+						{ label }
+					</label>
+					<select
+						className="wc-blocks-components-select__select"
+						id={ inputId }
+						size={ size !== undefined ? size : 1 }
+						onChange={ selectOnChange }
+						value={ value }
+						aria-invalid={
+							error?.message && ! error?.hidden ? true : false
+						}
+						aria-errormessage={ validationErrorId }
+						{ ...restOfProps }
```

Comment:
also from combobox.

Reply:
Is there a reason for just copy pasting `...restOfProps`. I'm not sure I understand the approach of just copy/pasting everything from the old component. If something doesn't work perhaps we should figure out what props are not being passed and just explicitly pass them.

There is a bit of a tendency higher up in the component tree to pass down all props and you get react warnings because some are not props that are legal on the html component.

Reply:
`...restOfProps` was already there in the select so it's not new addition, I only brought back `aria-invalid` and `aria-errormessage`, those are not supported by the old combobox but supported by the new select.